src/midi.rs

```diff
@@ -83,54 +95,82 @@ fn get_ports(midi_in: &MidiInput, print: bool) -> Result<std::vec::Vec<midir::Mi
 pub struct MidiConnection {
     port: MidiInputPort,
     port_name: String,
-    midi_in: Option<MidiInput>
+    midi_in: Option<MidiInput>,
+}
+
+fn port_prompt(ports: &Vec<MidiInputPort>, re_ask: Option<()>) -> usize {
+    let mut choice: usize = 0;
+    if let None = re_ask {
+        println!("Please choose your midi input.");
+    }
+    match read_input().expect("Couldn't read input.").parse::<usize>() {
+        Ok(c) => {
+            if c < ports.len() {
+                println!("You have chosen {}", c);
+                choice = c;
+            } else {
+                println!("Please enter a valid number of a port.");
+                port_prompt(ports, Some(()));
+            }
+        }
+        Err(_) => {
+            println!("Please enter a number of the port you want to choose instead.");
```

Comment:
```suggestion
            println!("{}: Not a valid port number", c);
```
I would go with the exact same message, keeps it simple.
If they're different, you get a different error message for `-1` and `99` which seems wonky.. :)

Reply:
You might be right I agree :)